docs/content/how-to/aws/create-aws-hosted-cluster-autonode.md

```diff
@@ -0,0 +1,452 @@
+---
+title: Create AutoNode hosted clusters powered by Karpenter
+---
+
+HyperShift AutoNode (Powered by Karpenter) is ...(TODO)
+
+To create a hosted cluster with autoNode enabled the Hypershift Operator
+needs to be installed with the `--tech-preview-no-upgrade=true` flag.
+
+The Hypershift controller creates and manages the default [`EC2NodeClass`][ec2nodeclass]
+in the hosted cluster allowing you to deploy workloads based in
+your environment with [`NodePools`][nodepools].
+
+The steps describes how to install OpenShift workload cluster
+on AWS with AutoNode feature by Karpenter.
+
+[ec2nodeclass]: https://karpenter.sh/docs/concepts/nodeclasses/
+[nodepools]: https://karpenter.sh/docs/concepts/nodepools/
+
+## Prerequisites
+
+### Export environment variables
+
+```sh
+# AWS config
+export AWS_CREDS="$HOME/.aws/credentials"
+export REGION=us-east-1
+export AWS_REGION=$REGION
+
+# OpenShift credentials and configuration
+export CLUSTER_BASE_DOMAIN=devcluster.openshift.com
+export PULL_SECRET_FILE="${HOME}/.openshift/pull-secret-latest.json"
+export SSH_PUB_KEY_FILE=$HOME/.ssh/id_rsa.pub
+
+# Management cluster
+export CLUSTER_NAME=hyp-aws01
+export INSTALL_DIR=${PWD}/${CLUSTER_NAME}
+
+# Hypershift
+export OIDC_BUCKET_NAME="${CLUSTER_NAME}-oidc"
+```
+
+### Install hypershift CLI
+
+Clone and build Karpenter controller:
+
+```sh
+git clone https://github.com/openshift/hypershift.git && cd hypershift
+make build
+install  -m 0755 bin/hypershift ~/bin/hypershift
+```
+
+#### Setup Bucket for OIDC discovery documents
+
+```sh
+aws s3api create-bucket --bucket $OIDC_BUCKET_NAME
+aws s3api delete-public-access-block --bucket $OIDC_BUCKET_NAME
+echo '{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": "s3:GetObject",
+      "Resource": "arn:aws:s3:::${OIDC_BUCKET_NAME}/*"
+    }
+  ]
+}' | envsubst > ${CLUSTER_NAME}-oidc-workload-clusters_policy.json
+aws s3api put-bucket-policy --bucket $OIDC_BUCKET_NAME --policy file://${CLUSTER_NAME}-oidc-workload-clusters_policy.json
+```
+
+### Enable service-linked role
+
+Create Service Role to allow AWS services to launch spot instance in your account (one time per account):
+
+```sh
+aws iam create-service-linked-role --aws-service-name spot.amazonaws.com
+```
+
+### Create Karpenter IAM Role
+
+Woraround: Create IAM Role used by Karpenter Controller
+
+> FIXME: https://issues.redhat.com/browse/PODAUTO-302
+> TODO/FIXME: create iam role (see known issues section)
+> TODO/FIXME: add required policy depending of the karpenter version.
+> Are we exporting it in the CPO image?
```

Comment:
Let's keep the fixme with the link to the known issue and drop the others. I see creating the role as an admin task. For hypershift "the project" this can be automated by the cli, so it will be create when cli create cluster with known name e.g karpenter-role-infra-id